test/python/base/gf_dlr.py

```diff
@@ -105,8 +117,9 @@ def test_dlr_gfs_density(self):
 
         # block gf test
         Bgw = BlockGf(name_list=['1', '2'], block_list=[g_c, g_c], make_copies=True)
+        Bden = Bgw.density()
         for block, gw in Bgw:
-            np.testing.assert_almost_equal(density(gw), ref)
+            np.testing.assert_almost_equal(Bden[block], ref)
```

Comment:
```suggestion
            np.testing.assert_almost_equal(gw, ref)
```

Reply:
but that would defeat the purpose here. I wanted to call the density function first for the BlockGf in line 120 and the compare the block density array. Does this make sense?

Reply:
You are correct, I meant to suggest something along the lines of
```python
     for block, den in Bden:
            np.testing.assert_almost_equal(den, ref)
```

Reply:
ah yes that makes sense. I just updated the lines accordingly.